### ISSUE TITLE
[Feature Request] Fastest IP address problem v0.102.0

### ISSUE THREAD

**Issue by yatounoneko** [AdguardTeam/AdGuardHome#1679]
Awesome, i try this function really can choose the fast ip. I've been looking forward to this feature.

i saw #715 This is only for non-china DNS cases, but if add china DNS in list, will affect some china dns choose Contaminated IP(if the feedback sometime faster than to other IP,,,,like google)
C:\Users\XXX>nslookup google.com
服务器:  UnKnown
Address:  10.168.99.3

非权威应答:
名称:    google.com
Address:  8.7.198.46

..so i also need to set ip the GFW list in upstream DNS server to bypass the china DNS
_(Because i using some China DNS provider...update dns will very fast.. can improve my Website debug)_

like this: [/google.com/]10.168.99.4
I built the second AdGuard-Home to specialize in overseas website DNS query, the problem can be fixed. But this requires adding thousands of rules to the upstream DNS list....

May i know whether can add a list of domain names pass to a specified DNS query? Very grateful.

**Comment by lycor1s** [comment]
You can try this if you are using Adguard on OpenWRT, it can add the GFWLIST and set the DNS server for them.
https://github.com/rufengsuixing/luci-app-adguardhome
![image](https://user-images.githubusercontent.com/23312700/82120113-64d8e700-97b6-11ea-89de-6a36b90580c6.png)

**Comment by ameshkov** [comment]
@yatounoneko 

> will affect some china dns choose Contaminated IP

What kind of contaminated IPs?

Wouldn't it be better to filter out responses with these IP addresses instead?

**Comment by Mosney** [comment]
No, I suppose you misunderstand what TCP-speedtest do totally, it not designed to avoid DNS pollution, just for faster TCP connection, you need to find another way to achieve this purpose, such as DNS bypass, DNS2SOCKS, ChinaDNS,  or some proxy software trick.

In another hand, in your example, the most of polluted IP will be Twitter, Facebook, Softlayer or some abroad IP, they cannot fast enough to affect the result, and some polluted IP which doesn't exist such as 240.0.0.0/8 and they cannot complete TCP-speedtest check too. so it will NOT affect China users which pollute by GFW.

In conclusion, you cannot hope AGH (which is not designed for this) to solve DNS pollution.

**Comment by yatounoneko** [comment]
> @yatounoneko
> 
> > will affect some china dns choose Contaminated IP
> 
> **What kind of contaminated IPs?**
> 
> Wouldn't it be better to filter out responses with these IP addresses instead?

oh, **china DNS will reply the incorrect ip to some domain**. 

I know my request maybe not designed for AGH, just i hope the add the specify domain name list to extra specify DNS parsing only.

**Comment by Mosney** [comment]
Actually, DNS can be setting in line, you just need to ensure AGH in it, don't limit yourself into one resolver.
And I repeat I said above, only fix DNS pollution alone has no meaningful in China, if you use proxy clients property, DNS pollution should be fixed. AGH is AdGuard designed for block AD, tracking, or something else.
